pkg/apis/networking/v1/types.go

```diff
@@ -174,3 +175,149 @@ type NetworkList struct {
 
 	Items []Network `json:"items"`
 }
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +k8s:openapi-gen=true
+
+// RemoteCluster is a specification for a RemoteCluster resource
+type RemoteCluster struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   RemoteClusterSpec   `json:"spec"`
+	Status RemoteClusterStatus `json:"status"`
+}
+
+// RemoteClusterSpec is the spec for a RemoteCluster resource
+type RemoteClusterSpec struct {
+	// globally unique id
+	ClusterID uint32 `json:"cluster_id"`
```

Comment:
why not just use cluster name for globally unique id?

Reply:
I prefer using cluster name as a alias, and it can be easily modified. ClusterID is somewhat like a constant, and shouldn't   make change during runtime.

Reply:
> I prefer using cluster name as a alias, and it can be easily modified. ClusterID is somewhat like a constant, and shouldn't make change during runtime.

If use 'Name' filed in metav1.TypeMeta as cluster name, it's also unchangeable. And same thing happens to other cluster-scoped k8s object too, e.g., name of a 'Node' cannot be edited.

Reply:
I see. I will delete both clusterID and clusterName.